clients/src/test/java/org/apache/kafka/common/metrics/JmxReporterTest.java

```diff
@@ -16,25 +16,59 @@
  */
 package org.apache.kafka.common.metrics;
 
+import org.apache.kafka.common.MetricName;
 import org.apache.kafka.common.metrics.stats.Avg;
 import org.apache.kafka.common.metrics.stats.Total;
+import org.hamcrest.CoreMatchers;
 import org.junit.Test;
 
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+
 public class JmxReporterTest {
 
     @Test
     public void testJmxRegistration() throws Exception {
-        Metrics metrics = new Metrics();
-        try {
-            metrics.addReporter(new JmxReporter());
+        final JmxReporter reporter = new JmxReporter();
+        MetricName name;
+        try (Metrics metrics = new Metrics()) {
+            assertEquals(0, reporter.mbeansQty());
+            metrics.addReporter(reporter);
+            assertEquals(1, reporter.mbeansQty());
             Sensor sensor = metrics.sensor("kafka.requests");
-            sensor.add(metrics.metricName("pack.bean1.avg", "grp1"), new Avg());
-            sensor.add(metrics.metricName("pack.bean2.total", "grp2"), new Total());
+            name = metrics.metricName("pack.bean1.avg", "grp1");
+            sensor.add(name, new Avg());
+            assertEquals(2, reporter.mbeansQty());
+            assertEquals(1, reporter.getMBean(name).attributesQty());
+            name = metrics.metricName("pack.bean2.total", "grp2");
+            sensor.add(name, new Total());
+            assertEquals(3, reporter.mbeansQty());
+            assertEquals(1, reporter.getMBean(name).attributesQty());
             Sensor sensor2 = metrics.sensor("kafka.blah");
-            sensor2.add(metrics.metricName("pack.bean1.some", "grp1"), new Total());
-            sensor2.add(metrics.metricName("pack.bean2.some", "grp1"), new Total());
-        } finally {
-            metrics.close();
+            name = metrics.metricName("pack.bean1.some", "grp1");
+            sensor2.add(name, new Total());
+            assertEquals(3, reporter.mbeansQty());
+            assertEquals(2, reporter.getMBean(name).attributesQty());
+            name = metrics.metricName("pack.bean2.some", "grp1");
+            sensor2.add(name, new Total());
+            assertEquals(3, reporter.mbeansQty());
+            assertEquals(3, reporter.getMBean(name).attributesQty());
+            metrics.removeSensor(sensor2.name());
+            assertEquals(3, reporter.mbeansQty());
+            assertEquals(1, reporter.getMBean(name).attributesQty());
+        }
+        assertEquals(0, reporter.mbeansQty());
+        assertEquals(0, reporter.getMBean(name).attributesQty());
+    }
+
+    @Test
+    public void testGetMBeanWithNull() {
+        final JmxReporter reporter = new JmxReporter();
+        try {
+            reporter.getMBean(null);
+        } catch (RuntimeException e) {
+            assertThat(e, CoreMatchers.instanceOf(NullPointerException.class));
+            assertThat(e.getMessage(), CoreMatchers.is("Parameter 'metricName' is mandatory"));
```

Comment:
If using hamcrest matchers, I think it's better to use static imports to make them more concise.